sql: TestExplainRedact failed

sql.TestExplainRedact [failed](https://mesolite.cluster.engflow.com/invocations/default/8e596adb-28c1-4f48-9ae5-4da9bffdae26?testReportRun=1&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [5e6a8549fad7ec01af94671b59d0be46d2b36240](https://github.com/cockroachdb/cockroach/commits/5e6a8549fad7ec01af94671b59d0be46d2b36240):


```
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Catalog (0.02s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan/Memo
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Memo (0.02s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan
        --- FAIL: TestExplainRedact/ExplainAnalyze/Plan (0.14s)
=== RUN   TestExplainRedact/Explain
    --- FAIL: TestExplainRedact/Explain (0.51s)
=== RUN   TestExplainRedact/Explain/Vec/Shape
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Shape (0.01s)
=== RUN   TestExplainRedact/ExplainAnalyze
    --- FAIL: TestExplainRedact/ExplainAnalyze (0.60s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode
        --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode (0.16s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode/NoFlag
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode/NoFlag (0.03s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode/Verbose
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode/Verbose (0.02s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan/Verbose
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Verbose (0.02s)
=== RUN   TestExplainRedact/Explain/Vec/NoFlag
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/NoFlag (0.01s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode/Shape
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode/Shape (0.02s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan/Shape
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Shape (0.02s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan/Viz
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Viz (0.02s)
=== RUN   TestExplainRedact/Explain/Vec
        --- FAIL: TestExplainRedact/Explain/Vec (0.08s)
=== RUN   TestExplainRedact/Explain/Vec/Memo
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Memo (0.01s)
=== RUN   TestExplainRedact/Explain/Vec/Types
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Types (0.01s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan/Types
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Types (0.02s)
=== RUN   TestExplainRedact/Explain/Vec/Viz
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Viz (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainRedact.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41234

sql.TestExplainRedact [failed](https://tanzanite.cluster.engflow.com/invocations/default/c4af67d4-2dbf-4c05-9731-9b6abc4cfbc6?testReportRun=7&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cvc3FsOnNxbF90ZXN0) on master @ [9765a2cc0d7b4d6dacff07ea5668d1b59a36cfc2](https://github.com/cockroachdb/cockroach/commits/9765a2cc0d7b4d6dacff07ea5668d1b59a36cfc2):


```
            --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode/Shape (0.08s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode/Types
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode/Types (0.08s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode/Viz
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode/Viz (0.09s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan/Shape
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Shape (0.10s)
=== RUN   TestExplainRedact/Explain/Vec
        --- FAIL: TestExplainRedact/Explain/Vec (0.07s)
=== RUN   TestExplainRedact/Explain/Vec/NoFlag
    explain_test_util.go:132: encountered non-internal error: pq: unable to vectorize execution plan: unsupported processor core
        EXPLAIN (VEC, REDACT) CREATE STATISTICS __auto__ FROM defaultdb.public.seed WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-0.001ms'
        
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/NoFlag (0.01s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode
        --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode (0.54s)
=== RUN   TestExplainRedact/ExplainAnalyze/Plan/Catalog
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/Plan/Catalog (0.07s)
=== RUN   TestExplainRedact/Explain/Vec/Catalog
    explain_test_util.go:132: encountered non-internal error: pq: unable to vectorize execution plan: unsupported processor core
        EXPLAIN (VEC, CATALOG, REDACT) CREATE STATISTICS __auto__ FROM defaultdb.public.seed WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-0.001ms'
        
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Catalog (0.01s)
=== RUN   TestExplainRedact/Explain/Vec/Viz
    explain_test_util.go:132: encountered non-internal error: pq: unable to vectorize execution plan: unsupported processor core
        EXPLAIN (VEC, VIZ, REDACT) CREATE STATISTICS __auto__ FROM defaultdb.public.seed WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-0.001ms'
        
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Viz (0.01s)
=== RUN   TestExplainRedact/ExplainAnalyze/NoMode/NoFlag
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/ExplainAnalyze/NoMode/NoFlag (0.07s)
=== RUN   TestExplainRedact/Explain/Vec/Types
    explain_test_util.go:132: encountered non-internal error: pq: unable to vectorize execution plan: unsupported processor core
        EXPLAIN (VEC, TYPES, REDACT) CREATE STATISTICS __auto__ FROM defaultdb.public.seed WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-0.001ms'
        
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Types (0.01s)
=== RUN   TestExplainRedact/Explain/Vec/Verbose
    explain_test_util.go:132: encountered non-internal error: pq: unable to vectorize execution plan: unsupported processor core
        EXPLAIN (VEC, VERBOSE, REDACT) CREATE STATISTICS __auto__ FROM defaultdb.public.seed WITH OPTIONS THROTTLING 0.9 AS OF SYSTEM TIME '-0.001ms'
        
    explain_test_util.go:128: pq: internal error: only one partial statistic can be requested at a time
            --- FAIL: TestExplainRedact/Explain/Vec/Verbose (0.01s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExplainRedact.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/distsql_plan_stats.go
+++ b/pkg/sql/distsql_plan_stats.go
@@ -35,6 +35,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/stats"
 	"github.com/cockroachdb/cockroach/pkg/sql/stats/bounds"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
+	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/logtags"
@@ -291,7 +292,10 @@ func (dsp *DistSQLPlanner) createPartialStatsPlan(
 	// Partial stats collections on multiple columns create different plans,
 	// so we only support one requested stat at a time here.
 	if len(reqStats) > 1 {
-		return nil, errors.AssertionFailedf("only one partial statistic can be requested at a time")
+		return nil, unimplemented.NewWithIssue(
+			128904,
+			"cannot process multiple partial statistics requests at once",
+		)
 	}
 
 	reqStat := reqStats[0]

